### ISSUE TITLE
Understand notification requirements and limitations

### ISSUE THREAD

**Issue by joshblack** [carbon-design-system/carbon#5864]
We had an interesting discussion around our notifications where it came to light that interactive content inside of a notification would be considered a WCAG 2.1 AA violation. This issue is to determine the scope of behavior a notification can support and coordinate how to update documentation and code to reflect these updates.

Previous discussion highlights:

Here’s how WAI ARIA seems to define some of these terms:

**Dialog**
A dialog is a window overlaid on either the primary window or another dialog window.

**Modal dialog**
A specific variant of a Dialog where windows under it are inert. That is, users cannot interact with content outside an active dialog window. Inert content outside an active dialog is typically visually obscured or dimmed so it is difficult to discern, and in some implementations, attempts to interact with the inert content cause the dialog to close.

**Alert**
An alert is an element that displays a brief, important message in a way that attracts the user’s attention without interrupting the user’s task.

**Alert and Message Dialog**
An alert dialog is a modal dialog that interrupts the user’s workflow to communicate an important message and acquire a response. Examples include action confirmation prompts and error message confirmations.

**Taxonomy**

![image](https://user-images.githubusercontent.com/3901764/79357482-b5ce9480-7f05-11ea-83a9-dbf5a6c05b4d.png)

**Links & Resources**

- https://www.scottohara.me/blog/2019/07/08/a-toast-to-a11y-toasts.html
- https://adrianroselli.com/2020/01/defining-toast-messages.html

**Comment by tay1orjones** [comment]
With v11 we've introduced ActionableNotification and clarified usage of Toast and Inline Notifications
